File: JavaScript/7kyu/CategorizeNewMember.js
function openOrSenior(data) {
  return data.map(([age, handicap]) =>
    age >= 55 && handicap > 7 ? "Senior" : "Open"
  );
}

function openOrSenior_fifi(data) {
  return data.reduce((pV, cV, cI) => {
    var result = cV[0] >= 55 && cV[1] > 7 ? "Senior" : "Open";

    pV.push(result);

    return pV;
  }, []);
}

export { openOrSenior };

var result = openOrSenior([
  [45, 12],
  [55, 21],
  [19, -2],
  [104, 20],
]);

console.log(result);

File: Python/7kyu/CategorizeNewMember.py
def open_or_senior(data):
    return ["Senior" if(a >= 55 and h > 7) else "Open" for a, h in data]


def open_or_senior_fifi(data):
    return [category(age, handicap) for age, handicap in data]


def category(age, handicap):
    if age >= 55 and handicap > 7:
        return "Senior"

    return "Open"


result = open_or_senior([(45, 12), (55, 21), (19, -2), (104, 20)])
print(result)
